src/ui.c

```diff
@@ -887,6 +942,37 @@ static void BarUiEventcmdPrintSong (FILE * restrict stream,
 			);
 }
 
+/* let user pick one station
+ * @param app handle
+ * @param stations that should be listed
+ * @param prompt string
+ * @param called if input was not a number
+ * @param auto-select if only one station remains after filtering
+ * @return pointer to selected station or NULL
+ */
+void BarUiSelectFilter(BarApp_t *app)
```

Comment:
I don’t like this UI, to be honest. It requires me to go to a different menu to filter the station selection by type. The idea of station selection was that you can just type name/id of the station and filter through that. It’d be more fitting if we could add some operator(?) to `BarUiSelectStation`.